lib/utils/compose_ts.ts

```diff
@@ -1063,18 +1063,21 @@ async function performResolution(
 					if (!buildTask.buildStream) {
 						continue;
 					}
-					// Consume each task.buildStream in order to trigger the
-					// resolution events that define fields like:
-					//     task.dockerfile, task.dockerfilePath,
-					//     task.projectType, task.resolved
-					// This mimics what is currently done in `resin-builder`.
-					const clonedStream: Pack = await cloneTarStream(
-						buildTask.buildStream,
-					);
-					buildTask.buildStream = clonedStream;
-					if (!buildTask.external && !buildTask.resolved) {
+					let error: Error | undefined;
+					try {
+						// Consume each task.buildStream in order to trigger the
+						// resolution events that define fields like:
+						//     task.dockerfile, task.dockerfilePath,
+						//     task.projectType, task.resolved
+						// This mimics what is currently done in `resin-builder`.
+						buildTask.buildStream = await cloneTarStream(buildTask.buildStream);
+					} catch (e) {
+						error = e;
+					}
+					if (error || (!buildTask.external && !buildTask.resolved)) {
+						const cause = error ? `${error}\n` : '';
 						throw new ExpectedError(
-							`Project type for service "${buildTask.serviceName}" could not be determined. Missing a Dockerfile?`,
+							`${cause}Project type for service "${buildTask.serviceName}" could not be determined. Missing a Dockerfile?`,
```

Comment:
I've tested this by injecting a dummy `throw new Error('oops')` statement right after the call to `cloneTarStream`. The console output was:
```
$ ./bin/balena build ~/balena/balena-io-examples/balena-idling/ -f test-rpi
[Info]    No "docker-compose.yml" file found at "/Users/paulo/balena/balena-io-examples/balena-idling"
[Info]    Creating default composition with source: "/Users/paulo/balena/balena-io-examples/balena-idling"
[Info]    Building for armv7hf/raspberrypi3
[Build]   Building services...
[Build]   main Preparing...
[Info]    Docker Desktop detected (daemon architecture: "x86_64")
[Info]      Docker itself will determine and enable architecture emulation if required,
[Info]      without balena-cli intervention and regardless of the --emulated option.
[Build]   Built 1 service in 0 seconds
[Error]   Build failed.
Error: oops
Project type for service "main" could not be determined. Missing a Dockerfile?
```